### PR TITLE
Add Builder.io <=> Next.js example demonstrating SSR AMP content

### PR DESCRIPTION
**Pull request by mandx**
## Description

Extremely simple demo on how to use Builder and Next.js to power server-rendered AMP sites.

### CODE REVIEW

File: examples/next-js-amp-ssr/pages/[...slug].js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    }
+    if (page.testVariationName === undefined) {
+      delete page.testVariationName;
+    }
+  }
+
+  return { props: { builderPage: page }};
+};
+
+// Page.getInitialProps = async function getInitialProps({ req, res, asPath }) {
```
Comment (mrkoreye) [inline examples/next-js-amp-ssr/pages/[...slug].js:279]: Can we remove these comments or add more comments as to why they are here?

File: examples/next-js-amp-ssr/pages/[...slug].js [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+import React from 'react';
+import Head from 'next/head';
+import { useAmp } from 'next/amp';
+
+import { builder } from '@builder.io/react';
+
+// This is a thin wrapper that does some content post-processing
+// when server-side rendering and AMP content is requested.
+// It's API is the same as Builder.io's `BuilderPage` component.
```
Comment (mrkoreye) [inline examples/next-js-amp-ssr/pages/[...slug].js:11]:
```suggestion
// Its API is the same as Builder.io's `BuilderPage` component.
```

**Review by mrkoreye — APPROVED** [review]

**Review by steve8708 — APPROVED** [review]

**Comment by steve8708** [comment]
one request - AMP is always ssr so let's call this folder `next-amp` or `next-js-amp` - I have a preference for the shortest `next-amp` since we all know that `react` is `react-js` etc :D

**Comment by mandx** [comment]
@steve8708 I'm going with `next-js-amp` for the folder name, to follow the same pattern (there's already a `next-js` folder/example).

**Pull request merged**